### DISCUSSION TITLE
Can I annotate and save a selection and  without needing to create two HTML files?

### DISCUSSION THREAD

**Discussion by Yekutiel** [gildas-lormeau/SingleFile#1069] [Ideas]
My use case is typically as follows...

1. I decide I want to save almost all of the content I have read on a webpage, yet I would like to exclude some information at the top or the bottom of the webpage because I find it extraneous.
2. Therefore, I use my mouse and keyboard to select almost the entire webpage while avoiding to select the information I want to exclude at the top or the bottom of the page which I find extraneous.
1. I use SingleFile to save the selection as **foo1.html**, and then 
2. I open foo1.html, and then 
3. I use SingleFile to annotate foo1.html, and then
4. I use SingleFile to save foo1.html as **foo2.html.**

That's a little cumbersome. Of course I'd prefer to simply annotate and save a selection as foo1.html, yet I don't see how to do that.

I would prefer not to use SingleFile to manually remove external elements because that would be much too cumbersome

**Comment by gildas-lormeau** [comment]
Enable `Annotation editor > annotate the page before saving` in the options page of SingleFile. You can also override files by selecting `File name > file name conflict resolution > overwrite the existing file`.

**Comment by Yekutiel** [comment]
Thank you very much, that worked!

In the context menu, just as `Auto-save` has options, I suppose `Save selection` should have options, specifically, two options: `Save without annotations` and `Add annotations prior to saving`. 

<img width="603" alt="2022-12-06_20-17" src="https://user-images.githubusercontent.com/16394972/206087181-a59e8ff5-337c-496f-96f3-a5457a180bf1.png">

Also, I suggest you add `Preferences` to the context menu.

In case it matters, I'm running Linux Mint Cinnamon 20.3.

**Comment by gildas-lormeau** [comment]
There is an entry equivalent to `Add annotations prior to saving` (i.e. `Annotate and save the page...`) but it's hidden when the option I suggested is enabled. You can usually open the options page from the context menu of the extension icon in the extensions bar (instead of the page).

**Comment by Yekutiel** [comment]
>> You can usually open the options page from the context menu of the extension icon in the extensions bar (instead of the page).

As the screenshot in my previous comment demonstrates, that assertion is false.

**Reply by gildas-lormeau** [reply]
I'm sorry but it's a bug in your browser then. I recommend you to report it to the vendor. For example, in Firefox on my machine, the item in the menu is `Manage Extension`.

**Comment by Yekutiel** [comment]
>> I'm sorry but it's a bug in your browser then. 

Thanks for letting me know.  

I'm running Linux Mint Cinnamon 20.3. and Firefox 107.0.1. Are you running Linux on your machine?

**Reply by gildas-lormeau** [reply]
No, I did the test on macOS, Windows and Ubuntu/WSL (with X Window in Win 11). It's surprising you don't see this entry. There are other entries for your information, see below (screenshot of the macOS version).

<img width="296" alt="image" src="https://user-images.githubusercontent.com/396787/207198353-c774efa9-1461-4fc6-9942-7655de3d6d97.png">

**Reply by Yekutiel** [reply]
>> It's surprising you don't see this entry.

It's not surprising to me because you did not test it on Linux.

Weinberg’s Law states, "If builders built buildings the way programmers wrote programs, then the first woodpecker that came along would destroy civilization."

Failing to test, is planning to fail. Perhaps you might retort, "But, but, but... it's Mozilla's fault! Contact them!" No, my dude. It's your fault.

**Reply by gildas-lormeau** [reply]
Will you please stop that tone. I provide SingleFile for free to everyone, I provide the source code, I answer user requests and all this without asking you a single cent. Please, change your attitude. You have the right to contribute in a positive way.

**Reply by gildas-lormeau** [reply]
And, again, this is a bug in your browser. There is nothing I can do to fix it on my end. Please report the bug to the vendor.

![image](https://user-images.githubusercontent.com/396787/207288883-1fcf65a8-1392-4aa1-a6da-b1a804617f73.png)

**Reply by Yekutiel** [reply]
Stop making inane assertions.

**Reply by gildas-lormeau** [reply]
I am sorry if the user support is not convenient for you. This is just a small amateur project.  Please contact professionals, they will be happy to help you.

**Reply by AeliusSaionji** [reply]
Yekutiel is looking at the wrong context menu.
